airflow/include/tasks/extract/astro_forum_docs.py

```diff
@@ -49,6 +50,7 @@ def filter_cutoff_questions(questions_urls: list[str]) -> list[str]:
     for question_url in questions_urls:
         try:
             html_content = requests.get(question_url).content
+            time.sleep(1)
```

Comment:
for now it is fine but I think we should only sleep if we get a time limit error maybe a 429 status code

Reply:
Sounds good. Will create a new issue tracking this to make a `requests` client wrapper that auto retries with some backoff mechanism to avoid duplicate code everywhere when `request` is used